main/src/cgeo/geocaching/utils/MapMarkerUtils.java

```diff
@@ -91,6 +94,8 @@ public static CacheMarker getCacheMarker(final Resources res, final Geocache cac
             .append(showPin(cacheListType))
             .append(showFloppyOverlay(cacheListType))
             .append(assignedMarkers)
+            .append(cache.getTerrain())
+            .append(cache.getDifficulty())
```

Comment:
If we add more and more parameters to the HashMap, does caching makes sense at all? Would it make sense to add the DT value only to the HashMap, if the related setting is enabled?

Reply:
Caching definitly makes sense. A "standard" situation of viewing a list, then that list as map, then previewing the cache, then opening it would result in 4 times drawing the marker instead of one. So even if the icon is unique it'll still be beneficial.

I'll make those 2 criteria optional though as you suggested.